TSError: ⨯ Unable to compile TypeScript:

<img width="1536" alt="Screen Shot 2022-09-24 at 07 39 38" src="https://user-images.githubusercontent.com/2212006/192095978-de71d5f1-5940-4dc2-b757-c1917c57d0af.png">

```
glebbahmutov at GlebMercari  ~/git/test-todomvc-using-app-actions on master
$ npm run print-tests

> test-todomvc-using-app-actions@1.0.0 print-tests
> find-cypress-specs --names --tags

/Users/glebbahmutov/git/test-todomvc-using-app-actions/node_modules/ts-node/src/index.ts:859
    return new TSError(diagnosticText, diagnosticCodes, diagnostics);
           ^
TSError: ⨯ Unable to compile TypeScript:
cypress.config.ts:16:14 - error TS2591: Cannot find name 'require'. Do you need to install type definitions for node? Try `npm i --save-dev @types/node` and then add 'node' to the types field in your tsconfig.

16       return require('./cypress/plugins/index.js')(on, config)
                ~~~~~~~
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -41,6 +41,13 @@ jobs:
 
       - name: Load cypress.config.ts 🤘
         run: npm run demo-tests --prefix test-ts
+      - name: Test cypress.config.ts project
+        run: npm test --prefix test-ts
+
+      - name: Load cypress.config.ts with import keyword 🤘
+        run: npm run demo-tests --prefix test-ts-import
+      - name: Test cypress.config.ts project with import keyword
+        run: npm test --prefix test-ts-import
 
       - name: Unit tests 🧪
         run: npm test

--- a/README.md
+++ b/README.md
@@ -119,6 +119,29 @@ $ npx find-cypress-specs --names --tagged <tag1>,<tag2>,<tag3>,...
 
 If the project uses TypeScript and `cypress.config.ts` then this module uses [ts-node/register](https://github.com/TypeStrong/ts-node) to load the config and fetch the spec pattern.
 
+If you are using `import` keyword in your `cypress.config.ts` you might get an error like this:
+
+```
+import { defineConfig } from 'cypress';
+^^^^^^
+
+SyntaxError: Cannot use import statement outside a module
+```
+
+In that case, add to your `tsconfig.json` file the `ts-node` block:
+
+```json
+{
+  "ts-node": {
+    "compilerOptions": {
+      "module": "commonjs"
+    }
+  }
+}
+```
+
+See example in [bahmutov/test-todomvc-using-app-actions](https://github.com/bahmutov/test-todomvc-using-app-actions).
+
 **Tip:** read my blog post [Convert Cypress Specs from JavaScript to TypeScript](https://glebbahmutov.com/blog/cypress-js-to-ts/).
 
 ## Details

--- a/src/index.js
+++ b/src/index.js
@@ -32,10 +32,10 @@ function getConfigJs(filename) {
 }
 
 function getConfigTs(filename) {
-  require('ts-node/register')
-  const jsFile = path.join(process.cwd(), filename)
-  debug('loading Cypress config from %s', jsFile)
-  const definedConfig = requireEveryTime(jsFile)
+  require('ts-node/register/transpile-only')
+  const configFilename = path.join(process.cwd(), filename)
+  debug('loading Cypress config from %s', configFilename)
+  const definedConfig = requireEveryTime(configFilename)
   return definedConfig
 }
 

--- a/test-ts-import/.gitignore
+++ b/test-ts-import/.gitignore
@@ -1,0 +1,3 @@
+node_modules/
+cypress/videos/
+cypress/screenshots/

--- a/test-ts-import/cypress.config.ts
+++ b/test-ts-import/cypress.config.ts
@@ -1,0 +1,13 @@
+import { defineConfig } from 'cypress'
+
+module.exports = defineConfig({
+  e2e: {
+    // baseUrl, etc
+    supportFile: false,
+    fixturesFolder: false,
+    setupNodeEvents(on, config) {
+      // implement node event listeners here
+      // and load any plugins that require the Node environment
+    },
+  },
+})

--- a/test-ts-import/cypress/e2e/spec1.cy.ts
+++ b/test-ts-import/cypress/e2e/spec1.cy.ts
@@ -1,0 +1,3 @@
+it('has first test')
+
+it('has second test')

--- a/test-ts-import/cypress/e2e/spec2.cy.ts
+++ b/test-ts-import/cypress/e2e/spec2.cy.ts
@@ -1,0 +1,3 @@
+it('has first test')
+
+it('has second test')

--- a/test-ts-import/package.json
+++ b/test-ts-import/package.json
@@ -1,0 +1,17 @@
+{
+  "name": "test-ts",
+  "version": "1.0.0",
+  "description": "Test finding Cypress specs with TS config file",
+  "private": true,
+  "main": "index.js",
+  "scripts": {
+    "test": "cypress run",
+    "demo-tests": "DEBUG=find-cypress-specs node ../bin/find"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
+  "devDependencies": {
+    "typescript": "^4.8.3"
+  }
+}

--- a/test-ts-import/tsconfig.json
+++ b/test-ts-import/tsconfig.json
@@ -1,0 +1,18 @@
+{
+  "include": ["cypress.config.ts", "cypress/e2e/*.cy.ts"],
+  "compilerOptions": {
+    "lib": ["DOM", "ES2015"],
+    "module": "es6",
+    "target": "es6",
+    "strict": true,
+    "allowJs": true,
+    "noEmit": true,
+    "moduleResolution": "node",
+    "resolveJsonModule": true
+  },
+  "ts-node": {
+    "compilerOptions": {
+      "module": "commonjs"
+    }
+  }
+}

--- a/test-ts/tsconfig.json
+++ b/test-ts/tsconfig.json
@@ -1,3 +1,3 @@
 {
-  "files": ["**/*.ts"]
+  "include": ["cypress.config.ts", "cypress/e2e/*.cy.ts"]
 }

